purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -0,0 +1,365 @@
+     * Initiate a purchase with the given [PurchaseParams].
+     * Initialized with an [Activity] either a [Package], [StoreProduct], or [SubscriptionOption].
+     *
+     * If a [Package] or [StoreProduct] is used to build the [PurchaseParams], the [defaultOption] will be purchased.
+     * [defaultOption] is selected via the following logic:
+     *   - Filters out offers with "rc-ignore-offer" tag
+     *   - Uses [SubscriptionOption] with the longest free trial or cheapest first phase
+     *   - Falls back to use base plan
+     *
+     *   @params [purchaseParams] The parameters configuring the purchase. See [PurchaseParams.Builder] for options.
+     *   @params [callback] The PurchaseCallback that will be called when purchase completes.
+     */
+    fun purchase(
+        purchaseParams: PurchaseParams,
+        callback: PurchaseCallback,
+    ) {
+        purchasesOrchestrator.purchase(purchaseParams, callback)
+    }
+
+    /**
+     * Restores purchases made with the current Play Store account for the current user.
+     * This method will post all purchases associated with the current Play Store account to
+     * RevenueCat and become associated with the current `appUserID`. If the receipt token is being
+     * used by an existing user, the current `appUserID` will be aliased together with the
+     * `appUserID` of the existing user. Going forward, either `appUserID` will be able to reference
+     * the same user.
+     *
+     * You shouldn't use this method if you have your own account system. In that case
+     * "restoration" is provided by your app passing the same `appUserId` used to purchase originally.
+     * @param [callback] The listener that will be called when purchase restore completes.
+     */
+    fun restorePurchases(
+        callback: ReceiveCustomerInfoCallback,
+    ) {
+        purchasesOrchestrator.restorePurchases(callback)
+    }
+
+    /**
+     * Call close when you are done with this instance of Purchases
+     */
+    fun close() {
+        purchasesOrchestrator.close()
+    }
+
+    /**
+     * Call this when you are finished using the [UpdatedCustomerInfoListener]. You should call this
+     * to avoid memory leaks.
+     */
+    @Suppress("MemberVisibilityCanBePrivate")
+    fun removeUpdatedCustomerInfoListener() {
+        purchasesOrchestrator.removeUpdatedCustomerInfoListener()
+    }
+
+    /**
+     * Invalidates the cache for customer information.
+     *
+     * Most apps will not need to use this method; invalidating the cache can leave your app in an invalid state.
+     * Refer to https://rev.cat/customer-info-cache for more information on
+     * using the cache properly.
+     *
+     * This is useful for cases where purchaser information might have been updated outside of the
+     * app, like if a promotional subscription is granted through the RevenueCat dashboard.
+     */
+    fun invalidateCustomerInfoCache() {
+        purchasesOrchestrator.invalidateCustomerInfoCache()
```

Comment:
not needed

Reply:
I believe this one is available in iOS. But yeah, I think it's probably ok to remove it. CC @NachoSoto

Reply:
Just asked to see if it's been used 👀